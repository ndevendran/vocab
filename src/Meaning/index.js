import React from 'react';
import {Button} from 'react-bootstrap';

function Meaning({definition, hidden, onClick}) {
  return (
    <div>
      {hidden
       ? <Button className="showDefinition" variant="info" onClick={onClick}>Show Definition</Button>
       : <span>{definition}</span>
      }
    </div>
  );
}

export default Meaning;
