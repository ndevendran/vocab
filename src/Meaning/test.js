import React from 'react';
import ReactDOM from 'react-dom';
import Meaning from './Meaning';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const props = {
    definition: "test",
    hidden: false,
    onClick: () => true,
  }
  ReactDOM.render(<Meaning ...props />, div);
  ReactDOM.unmountComponentAtNode(div);
});
