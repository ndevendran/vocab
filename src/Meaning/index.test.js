import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Meaning from '../Meaning';


describe('Meaning', () => {
  const props = {
    definition: "test",
    hidden: false,
    onClick: () => true,
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Meaning {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has valid snapshot', () => {
    const component = renderer.create(
      <Meaning {...props} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
