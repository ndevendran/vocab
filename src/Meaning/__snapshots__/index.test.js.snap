// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meaning has valid snapshot 1`] = `
<div>
  <span>
    test
  </span>
</div>
`;
